node/src/components/consensus/era_supervisor.rs

```diff
@@ -166,21 +168,19 @@ where
             next_upgrade_activation_point: None,
             stop_for_upgrade: false,
             next_executed_height: 0,
+            is_initialized: false,
         };
 
-        let outcomes = era_supervisor.new_era(
-            EraId(0),
-            timestamp,
-            validators,
-            vec![], // no banned validators in era 0
-            0,      // hardcoded seed for era 0
-            genesis_start_time,
-            0, // the first block has height 0
-            genesis_state_root_hash,
-        );
-        let effects = era_supervisor
-            .handling_wrapper(effect_builder, &mut rng)
-            .handle_consensus_outcomes(EraId(0), outcomes);
+        let effects = effect_builder
+            .collect_switch_blocks(era_supervisor.iter_past_other(current_era, bonded_eras * 3))
```

Comment:
Was that always `bonded_eras * 3`?

Reply:
Here, yes. We need to initialize `2 * bonded_eras` in the past, and to initialize the earliest one, we need switch blocks going `bonded_eras` earlier.